EIPS/eip-4361.md

```diff
@@ -0,0 +1,313 @@
+    - Any other resolvable resources specified in [the documentation](https://docs.ens.domains/dapp-developer-guide/resolving-names#looking-up-other-resources).
+- If resolution of ENS data is performed, implementers SHOULD take precautions to preserve user privacy and consent, as their `address` could be forwarded to third party services as part of the resolution process.
+
+### Server-Side (Relying Party) Implementer Guidelines
+
+#### Verifying a signed `message`
+- The message MUST be checked for conformance to the ABNF above, checked against expected term values after parsing, and its signature must be verified.
+
+#### Creating sessions
+- Sessions MUST be bound to the `address` and not to further resolved resources that may change.
+
+#### Interpreting and resolving `resources`
+- The listed `resources` must be RFC 3986 URIs, but their interpretation is out of scope of this specification.
+- Implementers SHOULD ensure that that URIs are human-friendly when expressed in plaintext form.
+
+### Wallet Implementer Guidelines
+
+#### Verifying `message`
+- The full `message` MUST be checked for conformance to the ABNF above.
+- Wallet implementers SHOULD warn users if the substring `"wants you to sign in
+  with your Ethereum account"` appears anywhere in an EIP-191 message signing
+  request unless the message fully conforms to the format defined in EIP-4361.
+
+#### Verifying `domain` binding
+- Wallet implementers MUST prevent phishing attacks by matching on the `domain` term when processing a signing request. For example, when processing the message beginning with `"service.org wants you to sign in..."`, the wallet checks that the request actually originated from `service.org`.
+- The domain SHOULD be read from a trusted data source such as the browser window or over WalletConnect ([EIP-1328](https://eips.ethereum.org/EIPS/eip-1328)) sessions for comparison against the signing message contents.
+
+#### Creating Sign-In with Ethereum interfaces
+- Wallet implementers MUST display to the user the following terms from the Sign-In with Ethereum signing request by default and prior to signing, if they are present: `domain`, `address`, `statement`, and `resources`. Other present terms MUST also be made available to the user prior to signing either by default or through an extended interface.
+- Wallet implementers displaying a plaintext `message` to the user SHOULD require the user to scroll to the bottom of the text area prior to signing.
+- Wallet implementers MAY construct a custom Sign-In With Ethereum user interface by parsing the ABNF terms into data elements for use in the interface. The display rules above still apply to custom interfaces.
+
+#### Supporting internationalization (i18n)
+- After successfully parsing the message into ABNF terms, translation may happen at the UX level per human language.
+
+
+## Rationale
+
+### Requirements
+From the [RFP](https://notes.ethereum.org/@djrtwo/sign-in-with-ethereum-RFP):
```

Comment:
The rationale section should explain the reasoning behind certain decisions in the EIP. Linking to the RFP is not appropriate here.

Reply:
Addressed in https://github.com/ethereum/EIPs/pull/4361/commits/e59daacf95ecf9747b47921f7ce7b865eb139f75